### PR TITLE
FIXED #3560 - Now we log queries in one, and only one, log line

### LINKED ISSUE #3560: Queries logged in multiple lines.
#### Issue

SuiteCRM records one system action in more than one log line.
	

#### Expected Behavior
Each log line should represent one (and only one) system action.
A log line with a query should be recorded this way:

     Tue May 16 09:58:11 2017 [30922][1][INFO] Query: SELECT  alerts.*  , jt0.user_name modified_by_name , jt0.created_by modified_by_name_owner  , 'Users' modified_by_name_mod , jt1.user_name created_by_name , jt1.created_by created_by_name_owner  , 'Users' created_by_name_mod , jt2.user_name assigned_user_name , jt2.created_by assigned_user_name_owner  , 'Users' assigned_user_name_mod FROM alerts   LEFT JOIN  users jt0 ON alerts.modified_user_id=jt0.id AND jt0.deleted=0     AND jt0.deleted=0  LEFT JOIN  users jt1 ON alerts.created_by=jt1.id AND jt1.deleted=0     AND jt1.deleted=0  LEFT JOIN  users jt2 ON alerts.assigned_user_id=jt2.id AND jt2.deleted=0     AND jt2.deleted=0 where (alerts.assigned_user_id = '1' AND is_read != '1') AND alerts.deleted=0

#### Actual Behavior
The previous log line is recorded in more than one line:

     Tue May 16 09:58:11 2017 [30922][1][INFO] Query: SELECT  alerts.*  , jt0.user_name modified_by_name , jt0.created_by modified_by_name_owner  , 'Users' modified_by_name_mod , jt1.user_name created_by_name , jt1.created_by created_by_name_owner  , 'Users' created_by_name_mod , jt2.user_name assigned_user_name , jt2.created_by assigned_user_name_owner  , 'Users' assigned_user_name_mod FROM alerts   LEFT JOIN  users jt0 ON alerts.modified_user_id=jt0.id AND jt0.deleted=0

     AND jt0.deleted=0  LEFT JOIN  users jt1 ON alerts.created_by=jt1.id AND jt1.deleted=0

     AND jt1.deleted=0  LEFT JOIN  users jt2 ON alerts.assigned_user_id=jt2.id AND jt2.deleted=0

     AND jt2.deleted=0 where (alerts.assigned_user_id = '1' AND is_read != '1') AND alerts.deleted=0

#### Possible Fix
Remove the breaks lines from the queries before they are executed and registered in the log

#### Steps to Reproduce
Just run SuiteCRM and watch the log ;-)

#### Context

When you are debuging it is **VERY annoying** that the queries (and other system actions) are recorded in the log in several lines.
That's not a style thing it's a productive one... with queries in several lines using unix tools (cat, grep, awk, sort, wc, etc, etc) is a headache

#### Your Environment

* SuiteCRM Version used: Any
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)): Any
* Environment name and version (e.g. MySQL, PHP 7): Any
* Operating System and version (e.g Ubuntu 16.04): Any

### PR DESCRIPTION
**Pull request by Abuelodelanada**
## Description

This pull request fixes #3560
Now each log line represents one (and only one) system action.
Now a log line with a query is recorded this way:

     Tue May 16 09:58:11 2017 [30922][1][INFO] Query: SELECT  alerts.*  , jt0.user_name modified_by_name , jt0.created_by modified_by_name_owner  , 'Users' modified_by_name_mod , jt1.user_name created_by_name , jt1.created_by created_by_name_owner  , 'Users' created_by_name_mod , jt2.user_name assigned_user_name , jt2.created_by assigned_user_name_owner  , 'Users' assigned_user_name_mod FROM alerts   LEFT JOIN  users jt0 ON alerts.modified_user_id=jt0.id AND jt0.deleted=0     AND jt0.deleted=0  LEFT JOIN  users jt1 ON alerts.created_by=jt1.id AND jt1.deleted=0     AND jt1.deleted=0  LEFT JOIN  users jt2 ON alerts.assigned_user_id=jt2.id AND jt2.deleted=0     AND jt2.deleted=0 where (alerts.assigned_user_id = '1' AND is_read != '1') AND alerts.deleted=0

## Motivation and Context

When you are debuging it is **VERY annoying** that the queries (and other system actions) are recorded in the log in several lines.
That's not a style thing it's a productive one... with queries in several lines using unix tools (cat, grep, awk, sort, wc, etc, etc) is a headache

## How To Test This
Run SuiteCRM and watch the log.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [ ] My change requires a change to the documentation.
-  [x] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

File: include/database/DBManager.php [hunk 1]
```diff
@@ -4039,4 +4039,9 @@ public function enableKeys($tableName)
      * @return string
      */
     abstract public function getGuidSQL();
+
+
+    function removeLineBreaks($sql) {
```
Comment (PedroErnst) [inline include/database/DBManager.php]: Please state the visibility of the function explicitly (private/public)

Comment (Abuelodelanada) [reply include/database/DBManager.php]: @PedroErnst  Done!

**Comment by Abuelodelanada** [comment]
Hi @PedroErnst  is it possible to merge this enhancement? This is very userful for sysadmins and developers

**Comment by Mausino** [comment]
We also using this enhancement and is very useful "for digging the bugs" as said my developer  :D   

@Abuelodelanada the assessing process i thing, i'm not sure, go over @Dillon-Brown and @e-reeley

**Comment by pgorod** [comment]
The first step would be to resolve the conflicts... after that, I agree, this one would be nice to merge.

**Comment by Abuelodelanada** [comment]
It has been a long time since I made this PR.
I'll do the PR again, because synchronizing the branch hotfix is becoming more complex than desired

**Comment by samus-aran** [comment]
@Abuelodelanada Oh I was working through the conflicts just there and resolved them.... I committed the merge but I think you may have closed it whilst I was doing so.

**Comment by samus-aran** [comment]
Give us a shout out when ready with a new PR and we'll get it merged in :+1:

**Pull request closed without merging**